doc: style: Update breathe and Doxygen guide (#797)
NordicPlayground/fw-nrfconnect-nrf: doc/nrf/doc_styleguide.rst

.. note::
   The ``:cpp:enum:`` and ``:cpp:enumerator:`` domains do not generate a link

>    due to an `issue <https://github.com/michaeljones/breathe/issues/437>`_ with

Reviewer: All links should go into the links.txt file - that makes it easier to keep them up-to-date.

>    breathe. In order for those to link correctly, use the following::

Reviewer: ```suggestion
   breathe.
   As a workaround, use the following command::
```

		const struct event_type *type_id;
	};

References
==========

> To link to functions, enums or structs from Doxygen itself you can use the

Reviewer: ```suggestion
To link to functions, enums, or structs within Doxygen itself, use the
```
